src/core/qgsdatasourceuri.h

```diff
@@ -404,6 +404,8 @@ class CORE_EXPORT QgsDataSourceUri
     QString mSrid;
     //! Generic params store
     QMultiMap<QString, QString> mParams;
+    //! Connection info parameters stored as generic params
+    QStringList mConnectionInfoParams = { "session_role" };
```

Comment:
This is an ugly approach -- why is it needed here?

Reply:
I need to add `session_role` parameter to `connectionInfo` string. 

I choose this approach to only update 2 methods and to simplify defined new connection information parameters: 
* `connectionInfo` to include `session_role` parameter in the conection information
* `uri` to do not duplicate `session_role` parameter in the uri string

An other approach could be to defined a private class property `mSessionRole` like `mAuthConfigId`, `mUsername` or `mPassword`. With a new private class property,  I have to update 4 methods:
* `setParam`
* `hasParam`
* `param`
* `connectionInfo`

@nyalldawson If you think defined a new private class property is better, I can update my code.

Reply:
I have removed the ugly code.